### PR TITLE
ekii/Tabs Refactor

### PR DESCRIPTION
**Pull request by erinakii**
Saw that some tests weren't passing, and fixed. 1 test is still now passing and will continue to work on it. WIP

### CODE REVIEW

**Comment by edx-webhook** [comment]
Thanks for the pull request, @erinakii! I've created [OSPR-3623](https://openedx.atlassian.net/browse/OSPR-3623) to keep track of it in JIRA. JIRA is a place for product owners to prioritize feature reviews by the engineering development teams.

Feel free to add as much of the following information to the ticket:
- supporting documentation
- edx-code email threads
- timeline information ("this must be merged by XX date", and why that is)
- partner information ("this is a course on edx.org")
- any other information that can help Product understand the context for the PR

All technical communication about the code itself will still be done via the GitHub pull request interface. As a reminder, [our process documentation is here](http://edx-developer-guide.readthedocs.org/en/latest/process/overview.html).



We can't start reviewing your pull request until you've submitted a     [signed contributor agreement](https://open.edx.org/wp-content/uploads/2019/01/individual-contributor-agreement.pdf)     or indicated your institutional affiliation. Please see the [CONTRIBUTING](https://github.com/edx/edx-platform/blob/master/CONTRIBUTING.rst)     file for more information.

File: src/Tabs/index.jsx [hunk 1]
```diff
@@ -106,13 +99,71 @@ class Tabs extends React.Component {
...
 };
 
+// TODO: custom validator that ensures labels and panels are the same length
+
 export default Tabs;
+
+Tabs.Label = ({
+  labelId, panelId, label, index, activeTab, toggle,
+}) => {
+  const selected = index === activeTab;
```
Comment (adamstankiewicz) [inline src/Tabs/index.jsx]: nit: Renaming the `selected` variable to `isSelected` might make it more clear elsewhere that this is a boolean. Same for anywhere that may use the variable `selected`.

File: src/Tabs/index.jsx [hunk 2]
```diff
@@ -106,13 +99,71 @@ class Tabs extends React.Component {
...
 Tabs.propTypes = {
-  labels: PropTypes.oneOfType([
-    PropTypes.arrayOf(PropTypes.string),
-    PropTypes.arrayOf(PropTypes.element),
-  ]).isRequired,
-  children: PropTypes.arrayOf(PropTypes.element).isRequired,
+  tabs: PropTypes.arrayOf(PropTypes.objectOf(PropTypes.string)).isRequired,
 };
 
+// TODO: custom validator that ensures labels and panels are the same length
```
[the remark this answers is not in the document]
Comment (erinakii) [reply src/Tabs/index.jsx]: Agreed! I think I can definitely remove this.

File: src/Tabs/index.jsx [hunk 3]
```diff
@@ -106,13 +99,71 @@ class Tabs extends React.Component {
...
 }
 
-// TODO: custom validator that ensures labels and panels are the same length
 Tabs.propTypes = {
-  labels: PropTypes.oneOfType([
-    PropTypes.arrayOf(PropTypes.string),
-    PropTypes.arrayOf(PropTypes.element),
-  ]).isRequired,
-  children: PropTypes.arrayOf(PropTypes.element).isRequired,
+  tabs: PropTypes.arrayOf(PropTypes.objectOf(PropTypes.string)).isRequired,
```
[the remark this answers is not in the document]
Comment (erinakii) [reply src/Tabs/index.jsx]: Changed Tabs.propTypes to above!

File: src/Tabs/Tabs.test.jsx [hunk 4]
```diff
@@ -3,16 +3,16 @@ import { shallow } from 'enzyme';
...
-    'third',
-  ],
-  children: [
-    <div>first</div>,
-    <div>second</div>,
-    <div>third</div>,
-  ],
+  tabs: [{
+    label: 'First',
+    content: 'I am the first',
```
Comment (adamstankiewicz) [inline src/Tabs/Tabs.test.jsx]: Looks like you're using the key name of `content` here but `panel` elsewhere. I think `content` makes a little more sense than `panel` as it describes what it does a little more clearly. We should make sure we're using `content` everywhere `panel` is used.

File: .gitignore [hunk 5]
```diff
@@ -5,3 +5,4 @@ npm-debug.log
 coverage
 jest*
 dist
+.vscode
```
Comment (adamstankiewicz) [inline .gitignore]: nit: add a newline at the end of the file

File: src/Tabs/index.jsx [hunk 6]
```diff
@@ -106,13 +99,71 @@ class Tabs extends React.Component {
...
 };
 
+// TODO: custom validator that ensures labels and panels are the same length
+
 export default Tabs;
+
+Tabs.Label = ({
```
[the remark this answers is not in the document]
Comment (erinakii) [reply src/Tabs/index.jsx]: changes made to the latter suggestion!

File: src/Tabs/Tabs.test.jsx (hunk 4 above)
Comment (erinakii) [reply src/Tabs/Tabs.test.jsx]: Sounds good! Changed panels to contents except for the classNames

**Comment by erinakii** [comment]
@adamstankiewicz I think that some tests are failing. They are the Storyshots.test.js.snap. Should I be making changes to this test as well?

File: src/Tabs/index.jsx [hunk 7]
```diff
@@ -33,59 +31,63 @@ class Tabs extends React.Component {
...
-            active: selected,
-          })}
+          className={classNames('btn nav-link nav-item', {
+          active: isSelected,
+        })}
         >
           {label}
-        </Button>
+        </button>
+
```
Comment (Tj-Tracy) [inline src/Tabs/index.jsx]: nitpick: unneeded newline.

Comment (erinakii) [reply src/Tabs/index.jsx]: thanks! Removed!

**Pull request merged**